### PR TITLE
feat:ticket statistic

### PR DESCRIPTION
**Pull request by tolehoai**

### CODE REVIEW

File: src/Event/BookingEvent.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+use App\Entity\Ticket;
+
+class BookingEvent
+{
+
+    public function __construct(Ticket $ticket)
+    {
+    }
+}
```
Comment (ntsanq) [inline src/Event/BookingEvent.php]: huh? What event is this? What do you get by this event?

File: src/EventSubscriber/BookingSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        $ticketsStatistic = new TicketsStatistic();
+        if (empty($ticketOfDay)) {
+            $ticketsStatistic->setDate($date);
+            $ticketsStatistic->setCancel(1);
+            $this->ticketsStatisticRepository->addTicketsStatistic($ticketsStatistic, true);
+        } else {
+            $this->ticketsStatisticRepository->updateSuccessStatistic($ticketOfDay[0]->getId(), $ticketOfDay[0]->getSuccess(), $ticketOfDay[0]->getDate());
+        }
+    }
+}
```
Comment (ntsanq) [inline src/EventSubscriber/BookingSubscriber.php]: huh?

File: src/Repository/TicketRepository.php [hunk 3]
```diff
@@ -107,4 +107,6 @@ private function addOrder($ticket)
 
         return $ticket;
     }
+
+
 }
```
Comment (ntsanq) [inline src/Repository/TicketRepository.php:112]: format? huh?

File: src/Repository/TicketsStatisticRepository.php [hunk 4]
```diff
@@ -16,8 +21,44 @@
...
+        $query = $this->ticketStatistic
+            ->update()
+            ->set('tkst.success', ':success')
+            ->setParameter('success', $successNumber + 1)
+            ->where('tkst.date=' . '\'' . $this->dateTimeToDate($cancelDate) . '\'');
+
+        return $query->getQuery()->execute();
+    }
+
+}
```
Comment (ntsanq) [inline src/Repository/TicketsStatisticRepository.php]: huh?

File: src/Service/StripeService.php [hunk 5]
```diff
@@ -85,10 +85,10 @@ public function refund(RefundRequest $refundRequest): ?Ticket
         $ticket = $this->ticketRepository->findOneBy(['paymentId' => $refundRequest->getPaymentId()]);
         $this->ticketService->cancel($ticket);
 
-        $this->stripe->refunds->create([
-            'payment_intent' => $refundRequest->getPaymentId(),
-            'amount' => $ticket->getTotalPrice()*100
-        ]);
+//        $this->stripe->refunds->create([
```
Comment (ntsanq) [inline src/Service/StripeService.php]: turn my code on

File: src/Event/BookingEvent.php (hunk 1 above)
Comment (tolehoai) [reply src/Event/BookingEvent.php]: I think I should drink some coffee

File: src/EventSubscriber/BookingSubscriber.php (hunk 2 above)
Comment (tolehoai) [reply src/EventSubscriber/BookingSubscriber.php]: My mistake

File: src/Repository/TicketsStatisticRepository.php (hunk 4 above)
Comment (tolehoai) [reply src/Repository/TicketsStatisticRepository.php]: This is EOL

**Review by ntsanq — APPROVED** [review]
OK

**Pull request merged**